File: Easy/26. Remove Duplicates from Sorted Array/Solution.js
// Solution # 1
const removeDuplicates = function (nums) {
    let uniqueCount = 1; // Initialize the count of unique elements.

    for (let i = 1; i < nums.length; i++) {
        if (nums[i] !== nums[i - 1]) {
            // If the current element is different from the previous element, it's a new unique element.
            nums[uniqueCount] = nums[i]; // Move the unique element to the next position.
            uniqueCount++; // Increment the count of unique elements.
        }
    }

    return uniqueCount;
};

// Solution # 2
function removeDuplicates(nums) {
    // Use filter to create a new array with unique elements
    const uniqueArray = nums.filter((value, index) => nums.indexOf(value) === index);

    // Copy the unique elements back to the original array
    for (let i = 0; i < uniqueArray.length; i++) {
        nums[i] = uniqueArray[i];
    }
    return uniqueArray.length;
}

File: Easy/26. Remove Duplicates from Sorted Array/Solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int c = 0;
        if (nums.size() == 0) {
            return 0;
        }
        for (int i = 0; nums[i] != nums[nums.size() - 1]; i++) {
            int curElm = nums[i], nuOfA = 0, pos = i + 1;
            while (pos < nums.size()) {
                if (curElm == nums[pos]) {
                    nuOfA++;
                } else {
                    break;
                }
                pos++;
            }
            if (nuOfA > 0) {
                shiftByNum(nums, nuOfA, i);
            }
            c++;
        }
        
        return c + 1;
    }

private:
    void shiftByNum(vector<int>& nums, int by, int pos) {
        for (int i = pos; i + by < nums.size(); i++) {
            nums[i] = nums[i + by];
        }
    }
};

